tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -248,9 +256,21 @@ private void InstrumentationGateway_AddHeadersResponseTags(object sender, Instru
         private void Report(ITransport transport, Span span, string resultData, bool blocked)
         {
             span.SetTag(Tags.AppSecEvent, "true");
-            var samplingPirority = _settings.KeepTraces
-                                       ? SamplingPriority.UserKeep : SamplingPriority.AutoReject;
-            span.SetTraceSamplingPriority(samplingPirority);
+
+            if (rateLimiterCounter < _settings.TraceRateLimit)
+            {
+                span.SetTraceSamplingPriority(_settings.KeepTraces ? SamplingPriority.UserKeep : SamplingPriority.AutoReject);
```

Comment:
While you're in here, do you mind adding a comment to clarify the use of `AutoReject`? I find the setting name confusing so I was already going to add this comment in another PR I'm working on. Maybe something like:
```
// NOTE: DD_APPSEC_KEEP_TRACES=false means "drop all traces by setting AutoReject".
// It does _not_ mean "stop setting UserKeep (do nothing)". It should only be used for testing.
```

Reply:
yes! fixed in 8a1b7d7cfb8567399adc367e4bb2c284b1a35ea7

Reply:
Thanks!